### DISCUSSION TITLE
Send defer model with dispatch

### DISCUSSION THREAD

**Discussion by mhgolij** [livewire/livewire#7860] [1. Help] [answered]
In Livewire 2, when an event was called, all defer values were sent to the component. However, in Livewire 3, this feature no longer exists. Consequently, in the provided code, the name value is not sent to the component and does not change. It's worth noting that the same code worked correctly using emit in Livewire 2.
```html
<input class="form-control" id="name" type="text" wire:model="name">
<button id="saveButton" class="btn btn-success">save </button>
@script
<script>
    $('#saveButton').click(function() {
    	// Some Code 
         $wire.dispatchSelf('save')
      }
    });
</script>
@endscript
```

**Comment by danie-ramdhani** [comment, marked as answer]
works for me....

```php
    #[\Livewire\Attributes\On('save')]
    public function save()
    {
        dd($this->name);
    }
```

```blade
    <input class="form-control" id="name" type="text" wire:model="name">
    <button class="btn btn-success" type="button" x-on:click="$wire.$dispatchSelf('save')">save</button>
```

form version:

```blade
    <form wire:submit="save">
        <input class="form-control" id="name" type="text" wire:model="name">
        <button class="btn btn-success" type="submit">save</button>
    </form>
```

**Comment by mhgolij** [comment]
> works for me....
> 
> ```
>     #[\Livewire\Attributes\On('save')]
>     public function save()
>     {
>         dd($this->name);
>     }
> ```
> 
> ```
>     <input class="form-control" id="name" type="text" wire:model="name">
>     <button class="btn btn-success" type="button" x-on:click="$wire.$dispatchSelf('save')">save</button>
> ```
> 
> form version:
> 
> ```
>     <form wire:submit="save">
>         <input class="form-control" id="name" type="text" wire:model="name">
>         <button class="btn btn-success" type="submit">save</button>
>     </form>
> ```

Thank you very much, I wonder why this issue worked with JavaScript event listeners in the previous version, but it must be Alpine in this version.
